src/widgetastic_patternfly/__init__.py

```diff
@@ -660,6 +660,61 @@ def __repr__(self):
         return '<Tab {!r}>'.format(self.tab_name)
 
 
+class TabWithDropdown(Tab):
+    """Tab with a dropdown. Variant that always takes the sub item name in the select().
+
+    Does not support automatic reveal of the tab upon access as the default dropdown item is not
+    specified.
+    """
+    def is_dropdown(self):
+        return 'dropdown' in self.parent_browser.classes(self.TAB_LOCATOR)
+
+    def is_open(self):
+        return 'open' in self.parent_browser.classes(self.TAB_LOCATOR)
+
+    def open(self):
+        if not self.is_open():
+            self.logger.info('opened the tab %s', self.tab_name)
+            self.click()
+
+    def close(self):
+        if self.is_open():
+            self.logger.info('closed the tab %s', self.tab_name)
+            self.click()
+
+    def select(self, sub_item):
+        if not self.is_dropdown():
+            raise TypeError('{} is not a tab with dropdown and CHECK_IF_DROPDOWN is True')
+        self.open()
+        parent = self.parent_browser.element(self.TAB_LOCATOR)
+        self.logger.info('clicking the sub-item %r', sub_item)
+        self.parent_browser.click(
+            './ul/li[normalize-space(.)={}]'.format(quote(sub_item)),
+            parent=parent)
+
+    def child_widget_accessed(self, widget):
+        """Nothing. Since we don't know which sub_item."""
+
+    def __repr__(self):
+        return '<TabWithDropdown {!r}>'.format(self.tab_name)
+
+
+class TabWithDropdownDefault(TabWithDropdown):
```

Comment:
How about  a bit different naming, like TabWithDropdown/TabWithDropdownDefault -> GenericTabWithDropdown/TabWithDropdown or smth else? It's just sounds very unlikely to me that user will commonly use the class where you have to specify sub_item for each `select()` call, so i'd use "TabWithDropdown" name for the one which will actually be used - it seems kinda weird to use suffix 'Default' every time.

Reply:
Sure, I can rename them.